splice_machine/src/main/java/com/splicemachine/derby/impl/sql/execute/actions/CreateTriggerConstantOperation.java

```diff
@@ -257,9 +257,19 @@ public void executeConstantAction(Activation activation) throws StandardExceptio
         ** our trigger spses lest we invalidate them just
         ** after creating them.
         */
-        dm.invalidateFor(triggerTable, DependencyManager.CREATE_TRIGGER, lcc);
-
-        notifyMetadataChange(tc, ProtoUtil.createTrigger(((SpliceTransactionManager) tc).getActiveStateTxn().getTxnId(), (BasicUUID) this.tableId));
+        try {
+            dm.invalidateFor(triggerTable, DependencyManager.CREATE_TRIGGER, lcc);
+        }  catch (Throwable e) {
+           if (!lcc.isCloningData()) {
+               throw e;
+           }
+           else {
+               SpliceLogUtils.warn(LOG, "Get an exception when trying to invalidate dependencies", e);
+           }
+        }
+        if (!lcc.isCloningData()) {
+            notifyMetadataChange(tc, ProtoUtil.createTrigger(((SpliceTransactionManager) tc).getActiveStateTxn().getTxnId(), (BasicUUID) this.tableId));
```

Comment:
It looks like we should also protect calls to notifyMetadataChange with the "if (!lcc.isCloningData())" check in these classes:  FkJobSubmitter, CreateRoleConstantOperation, CreateViewConstantOperation, GrantRevokeConstantOperation

I'm assuming foreign keys, roles, privileges and views are also cloned?

Reply:
roles and privileges are not cloned

Reply:
Done for FkJobSubmitter

Reply:
@jyuanca What about views?